### PR TITLE
DIS-131 Add the constraint checks for creating a post/reply

### PR DESCRIPTION
**Pull request by jogura**
* Checks the status of the thread: isHeld, isViewable, isDeleted and if exists
* Create unit tests to cover new constraints

@jimmordino @armonr

### CODE REVIEW

File: service/discussion/src/main/java/com/wikia/discussionservice/services/AuthorizationService.java [hunk 1]
```diff
@@ -116,7 +117,48 @@
...
+    }
+
+    if (forumThread.getIsHeld()) {
+      LOGGER.info(String.format("An attempt to add post to a thread that does is held "
+                                    + "was performed by user id: %s site id: %s thread id: %s",
+                                user.getUserId(), siteId, threadId));
+      throw new DiscussionServiceException(
+          String.format("Unable to create the post on thread id: %s. "
+                            + "The thread is held.", threadId));
+    }
```
Comment (jimmordino) [inline service/discussion/src/main/java/com/wikia/discussionservice/services/AuthorizationService.java]: If you're going to check ```isHeld``` (a post-MVP feature), you also need to check ```isEditable```.  Threads can be locked/unlocked, which would change the value of this flag.

Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/services/AuthorizationService.java]: I thought `isEditable` was just on the thread's attributes (titles, etc). Does it extend to the posts?

Comment (jimmordino) [inline service/discussion/src/main/java/com/wikia/discussionservice/services/AuthorizationService.java]: it's everything about the thread - you can't modify the post list if the thread is not editable

**Comment by jimmordino** [comment]
:+1: , pending tests

**Pull request merged**